homeassistant/components/keyboard_remote.py

```diff
@@ -112,16 +113,42 @@ def __init__(self, hass, device_descriptor, key_value):
         self.stopped = threading.Event()
         self.hass = hass
         self.key_value = key_value
+        self.device_descriptor = device_descriptor
 
     def run(self):
         """Main loop of the KeyboardRemote."""
-        from evdev import categorize, ecodes
+        from evdev import categorize, ecodes, InputDevice
         _LOGGER.debug('KeyboardRemote interface started for %s', self.dev)
 
         self.dev.grab()
+        keyboard_connected = True
 
         while not self.stopped.isSet():
-            event = self.dev.read_one()
+
+            # Is keyboard still there?
+            keyboard_still_connected = os.path.exists(self.device_descriptor)
+
+            # still disconnected
+            if not keyboard_connected and not keyboard_still_connected:
+                continue
```

Comment:
Yeah, not 100% but easily a +40% on my server.
I added a sleep(.1) at each cycle: now there's no load increase, as far as I can tell.